ostd/src/arch/x86/allocator.rs

```diff
@@ -54,3 +57,28 @@ pub(super) fn construct_io_mem_allocator_builder() -> IoMemAllocatorBuilder {
     // SAFETY: The range is guaranteed not to access physical memory.
     unsafe { IoMemAllocatorBuilder::new(ranges) }
 }
+
+/// Initializes the allocatable PIO area outside OSTD based on the x86-64 port distribution map.
+pub(super) fn construct_io_port_allocator_builder() -> IoPortAllocatorBuilder {
```

Comment:
I suggest renaming this file from `allocator.rs` to `io.rs`. People may (mistakenly) assume that this so-called allocator module is related to memory allocator; but it is _not_. In addition, renaming this file to `src/arch/x86/io.rs` matches the `src/io` file.